### PR TITLE
Make first part of section name matter

### PR DESCRIPTION
**Pull request by HansOlsson**
The basic idea should be to "summarize" the section so that you can understand its contents from that short text - to me that makes the index more usable than trying to support the overly long sections.

Some specifics:

- Start with the important stuff
- Avoid unnecessary words:
  - In particular avoid repeating the section-name as start of the sub-section; as it doesn't work well in the table-of-contents. The names will work even if we were referencing using section name instead of numbers.
  - Avoid semantic restrictions in the section heading, especially if that makes it too long. E.g., when looking for `A^b` it suffices to say (scalar) exponentiation of matrices, and then you can read it and see that `A` must be numeric and square.

As proposed in #2825
Not yet complete.

### CODE REVIEW

**Comment by HansOlsson** [comment]
This should be ready now. There might be additional improvement later, but this should make it so that you don't need to see the entire section-name to navigate.

**Review by sjoelund — APPROVED** [review]

File: chapters/classes.tex [hunk 1]
```diff
@@ -729,7 +729,7 @@ \subsection{Short Class Definitions}\label{short-class-definitions}
 \end{lstlisting}
 \end{example}
 
-\subsection{Restriction on combining base classes and other elements}\label{restriction-on-combining-base-classes-and-other-elements}
+\subsection{Combining base classes and other elements}\label{restriction-on-combining-base-classes-and-other-elements}\label{combining-base-classes-and-other-elements}
```
Comment (eshmoylova) [inline chapters/classes.tex:732]: Capitalization:
```suggestion
\subsection{Combining Base Classes and Other Elements}\label{restriction-on-combining-base-classes-and-other-elements}\label{combining-base-classes-and-other-elements}
```
There is also uncapitalized "Unspecified enumeration" on line 1653 of this file/ But GutHub does not allow me to comment there because it is not part of this change.

File: chapters/equations.tex [hunk 2]
```diff
@@ -883,7 +883,7 @@ \subsection{The Number of Equations Needed for Initialization}\label{the-number-
 As noted in \cref{initialization-initial-equation-and-initial-algorithm} a tool may add/remove initial equations to fulfill this requirement, if appropriate diagnostics are given.
 \end{example}
 
-\subsection{Recommended selection of start values}\label{recommended-selection-of-start-values}
+\subsection{Start value recommended Priority}\label{recommended-selection-of-start-values}\label{start-value-recommended-priority}
```
Comment (eshmoylova) [inline chapters/equations.tex:886]: Shouldn't it be capitalized?
```suggestion
\subsection{Start Value Recommended Priority}\label{recommended-selection-of-start-values}\label{start-value-recommended-priority}
```

File: chapters/functions.tex [hunk 3]
```diff
@@ -2579,7 +2579,7 @@ \subsubsection{Input Parameters, Function Value}\label{input-parameters-function
 \end{lstlisting}
 \end{example}
 
-\subsubsection{Arbitrary Placement of Output Parameters, No External Function Value}\label{arbitrary-placement-of-output-parameters-no-external-function-value}
+\subsubsection{Arbitrary Placement of Output Parameters, No External Function Value}\label{arbitrary-placement-of-output-parameters-no-external-function-value}\label{both-function-value-and-output-variable}
```
Comment (eshmoylova) [inline chapters/functions.tex:2582]: Is this label in the right place? The subsubsection called Both Function Value and Output Variable is on line 2612.

File: chapters/statements.tex [hunk 4]
```diff
@@ -26,7 +26,7 @@ \subsection{Initial Algorithm Sections}\label{initial-algorithm-sections}
 See \cref{initialization-initial-equation-and-initial-algorithm} for a description of both initial algorithm sections and
 initial equation sections.
 
-\subsection{Execution of an algorithm in a model}\label{execution-of-an-algorithm-in-a-model}
+\subsection{An algorithm in a model}\label{execution-of-an-algorithm-in-a-model}\label{an-algorithm-in-a-model}
```
Comment (eshmoylova) [inline chapters/statements.tex:29]: Are algorithm and model considered to be keywords in this case? If they are not, they should be capitalized. In the subsection title "Initial Algorithm" are not treated as keywords but in Section 8.6 title initial equation and initial algorithm are treated as keywords. At least, I assume they are because they are not capitalized.

File: chapters/statements.tex [hunk 5]
```diff
@@ -64,7 +64,7 @@ \subsection{Execution of an algorithm in a model}\label{execution-of-an-algorith
 This is usually the case, except for algorithms with \lstinline!when!-statements, and especially for initial algorithms.
 \end{nonnormative}
 
-\subsection{Execution of the algorithm in a function}\label{execution-of-the-algorithm-in-a-function}
+\subsection{The algorithm in a function}\label{execution-of-the-algorithm-in-a-function}\label{the-algorithm-in-a-function}
```
Comment (eshmoylova) [inline chapters/statements.tex:67]: Same question as above: are algorithm and function are used as keywords?

File: chapters/statements.tex [hunk 6]
```diff
@@ -392,7 +392,7 @@ \subsection{When-Statements}\label{when-statements}
 Merging the \lstinline!when!-statements can lead to less efficient code and different models with different behavior depending on the order of the assignment to \lstinline!y1! and \lstinline!y3! in the algorithm.
 \end{example}
 
-\subsubsection{Restrictions on Where a When-statement may occur}\label{restrictions-on-where-a-when-statement-may-occur}
+\subsubsection{Where a When-statement may occur}\label{restrictions-on-where-a-when-statement-may-occur}\label{where-a-when-statement-may-occur}
```
Comment (eshmoylova) [inline chapters/statements.tex:395]: Hmm, here "when" should be a keyword in "when-statement" but it is capitalized while "may occur" is not.

File: chapters/statements.tex [hunk 7]
```diff
@@ -150,7 +150,7 @@ \subsubsection{Assignments from Called Functions with Multiple Results}\label{as
 Also see \cref{simple-equality-equations} regarding calling functions with multiple results within equations.
 \end{nonnormative}
 
-\subsubsection{Restrictions on assigned variables}\label{restrictions-on-assigned-variables}
+\subsubsection{Assigned variables - Restrictions}\label{restrictions-on-assigned-variables}\label{assigned-variables-restrictions}
```
Comment (eshmoylova) [inline chapters/statements.tex:153]: Capitalization
```suggestion
\subsubsection{Assigned Variables - Restrictions}\label{restrictions-on-assigned-variables}\label{assigned-variables-restrictions}
```

File: chapters/statements.tex [hunk 8]
```diff
@@ -416,7 +416,7 @@ \subsubsection{Restrictions on Where a When-statement may occur}\label{restricti
 \end{lstlisting}
 \end{example}
 
-\subsubsection{Restrictions on Statements within When-Statements}\label{restrictions-on-statements-within-when-statements}
+\subsubsection{Statements within When-Statements}\label{restrictions-on-statements-within-when-statements}\label{statements-within-when-statements}
```
Comment (eshmoylova) [inline chapters/statements.tex:419]: Another inconsistent capitalization: "When-Statement" while on line 395 it is "When-statement"

**Review by eshmoylova — CHANGES_REQUESTED** [review]
There are some inconsistent capitalizations in the section names especially in cases where keyword are used. Sometimes they are capitalized, sometimes not. And one case where it looks like the label is applied to a wrong section. The comments are in the relevant places of the code. Two more suggestions that I could not place in the code.

In the Annotations chapter, there are a lot of sections that start with "Annotations for..." Maybe it can be dropped. There is already a precedence of that in "Annotations for Graphical Objects". The section name includes "Annotations for..." but subsection and subsubsection names do not.

In the Arrays chapter, there is "Array Constructor with Iterators" with subsubsections "Array Constructor with  One Iterator" and "Array Constructor with Several Iterators". Maybe "Array" in the subsubsections can be dropped?

**Review by henrikt-ma — COMMENTED** [review]
It looks like we should have a discussion about redundant labels (`\label{foo}\label{bar}`) for cross references.  To me, it looks like a bad idea to introduce the habit of having redundant labels.  That said, I prefer the new and shorter labels, but at the same time I think it would be best if we could decouple the labels from the headings and in this regard it could make more sense to keep the old unwieldy labels.

**Comment by henrikt-ma** [comment]
> In the Annotations chapter, there are a lot of sections that start with "Annotations for..." Maybe it can be dropped. There is already a precedence of that in "Annotations for Graphical Objects". The section name includes "Annotations for..." but subsection and subsubsection names do not.
> 
> In the Arrays chapter, there is "Array Constructor with Iterators" with subsubsections "Array Constructor with One Iterator" and "Array Constructor with Several Iterators". Maybe "Array" in the subsubsections can be dropped?

I like those suggestions, especially the first one regarding annotations.

File: chapters/classes.tex [hunk 9]
```diff
@@ -71,7 +71,7 @@ \section{Double Declaration not Allowed}\label{double-declaration-not-allowed}
 \end{lstlisting}
 \end{example}
 
-\section{Declaration Order and Usage before Declaration}\label{declaration-order-and-usage-before-declaration}
+\section{Declaration Order - Usage before Declaration}\label{declaration-order-usage-before-declaration}\label{declaration-order-and-usage-before-declaration}
```
Comment (henrikt-ma) [inline chapters/classes.tex:74]: En-dash:
```suggestion
\section{Declaration Order -- Usage before Declaration}\label{declaration-order-usage-before-declaration}\label{declaration-order-and-usage-before-declaration}
```

Comment (henrikt-ma) [reply chapters/classes.tex:74]: How about even dropping the part after the dash entirely?
```suggestion
\section{Declaration Order}\label{declaration-order}\label{declaration-order-and-usage-before-declaration}
```

Comment (HansOlsson) [reply chapters/classes.tex:74]: Yes.

File: chapters/statements.tex (hunk 4 above)
Comment (HansOlsson) [reply chapters/statements.tex:29]: I would say non-keyword "algorithm" here. Will change

**Comment by HansOlsson** [comment]
> > In the Annotations chapter, there are a lot of sections that start with "Annotations for..." Maybe it can be dropped. There is already a precedence of that in "Annotations for Graphical Objects". The section name includes "Annotations for..." but subsection and subsubsection names do not.
> > In the Arrays chapter, there is "Array Constructor with Iterators" with subsubsections "Array Constructor with One Iterator" and "Array Constructor with Several Iterators". Maybe "Array" in the subsubsections can be dropped?
> 
> I like those suggestions, especially the first one regarding annotations.

Agreed, especially on annotations.

File: chapters/statements.tex (hunk 6 above)
Comment (HansOlsson) [reply chapters/statements.tex:395]: Will capitalize 'may occur'. I think capitalizing When-statement makes sense; as it's a compound and not the keyword on its own.

File: chapters/functions.tex (hunk 3 above)
Comment (HansOlsson) [reply chapters/functions.tex:2582]: oops - no

File: chapters/statements.tex (hunk 4 above)
Comment (HansOlsson) [reply chapters/statements.tex:29]: Will also change 'model' and 'function', even though keywords; as it is more about the concepts.

File: chapters/statements.tex (hunk 5 above)
Comment (HansOlsson) [reply chapters/statements.tex:67]: Will also change.

File: chapters/statements.tex (hunk 8 above)
Comment (HansOlsson) [reply chapters/statements.tex:419]: Hmm... Good point, I had to investigate and it seems that when you capitalize it as a title it should be 'When-Statement'. https://titlecaseconverter.com/blog/capitalizing-hyphenated-words/

**Comment by HansOlsson** [comment]
> It looks like we should have a discussion about redundant labels (`\label{foo}\label{bar}`) for cross references. To me, it looks like a bad idea to introduce the habit of having redundant labels. That said, I prefer the new and shorter labels, but at the same time I think it would be best if we could decouple the labels from the headings and in this regard it could make more sense to keep the old unwieldy labels.

We could remove the old ones. I kept them in case there's existing references to the document, but we can see that as less important, we could discuss that later.

However, the main goal with the labels was that you should be able to look at a section and construct the label automatically.

File: chapters/statements.tex (hunk 8 above)
Comment (HansOlsson) [reply chapters/statements.tex]: Should now be done.

File: chapters/statements.tex [hunk 10]
```diff
@@ -392,7 +392,7 @@ \subsection{When-Statements}\label{when-statements}
 Merging the \lstinline!when!-statements can lead to less efficient code and different models with different behavior depending on the order of the assignment to \lstinline!y1! and \lstinline!y3! in the algorithm.
 \end{example}
 
-\subsubsection{Restrictions on Where a When-statement may occur}\label{restrictions-on-where-a-when-statement-may-occur}
+\subsubsection{Where a When-statement May Occur}\label{restrictions-on-where-a-when-statement-may-occur}\label{where-a-when-statement-may-occur}
```
Comment (eshmoylova) [inline chapters/statements.tex:395]: Now we have For-Statements, If-Statements, even When-Statements as the subsection name. But here and in the changes below (lines 419, 434) they became When-statements.

**Review by eshmoylova — COMMENTED** [review]
Other things that are not changed by the PR but are relevant since we are trying to improve the look of it. Since there are no current code changes in the right placed GitHub does not allow me to comment in place.

On line 210 of annotations.tex, "Escape sequences" is also not capitalized consistently. Also in annotations.tex but cannot find the place in code (just looking at the generated table of contents): Extends-clause, User input, Mouse input, Edit input.

File: chapters/statements.tex (hunk 8 above)
Comment (eshmoylova) [reply chapters/statements.tex]: Looks good. Resolving.

File: chapters/statements.tex (hunk 6 above)
Comment (eshmoylova) [reply chapters/statements.tex]: Resolving.

File: chapters/statements.tex (hunk 5 above)
Comment (eshmoylova) [reply chapters/statements.tex]: Resolving

**Review by eshmoylova — COMMENTED** [review]
Ugh, those capitalizations are hard to find even when you are looking for them! Hopefully these are the last ones.
- "Edit input" on line 1100 in annotations.tex
- "Conversion rules" on line 1579 in annotations.tex
- "Unspecified enumeration" on line 1653 in classes.tex. I am not sure if here enumeration is supposed to be a keyword.
- "Field Functions using Inner/Outer" on line 262 in scoping.tex.
- "The class extends Redeclaration Mechanism" on line 651 in unheritance.tex. I am not sure about this one too. It looks odd the way it is. Maybe it should be _The "class extends" mechanism_? Saying _The Class Extends Mechanism_ doesn't quite look right either.
- "Synchronous Data-flow Principle and Single Assignment Rule" on line 530 of equations.tex. Also not sure if this the right way to capitalize.
- In Modelica Revision History there are a bunch of "Main changes in..." and a couple of "Main Changes in...:

**Comment by HansOlsson** [comment]
> Ugh, those capitalizations are hard to find even when you are looking for them! Hopefully these are the last ones.

Hope so as well.

> * "Field Functions using Inner/Outer" on line 262 in scoping.tex.

To me the current form looked good, but all styles say that verbs shall be capitalized so I changed it.

> * "The class extends Redeclaration Mechanism" on line 651 in unheritance.tex. I am not sure about this one too. It looks odd the way it is. Maybe it should be _The "class extends" mechanism_? Saying _The Class Extends Mechanism_ doesn't quite look right either.

I think the quotes is the right way.

> * "Synchronous Data-flow Principle and Single Assignment Rule" on line 530 of equations.tex. Also not sure if this the right way to capitalize.

I think 'Data-Flow'; and leave it for later.

**Review by eshmoylova — COMMENTED** [review]
Looks good, except this one was not updated:
> - "Conversion rules" on line 1579 in annotations.tex

**Comment by HansOlsson** [comment]
> 

Should be done now.

**Review by eshmoylova — APPROVED** [review]
Looks good
